File: framework/languages/java/samples/java/Bingo/Server/Play/src/main/java/systems/zlink/samples/bingo/server/play/infrastructure/zlink/spots/bingoroomspot/handlers/BingoWinnerEventHandler.java
package systems.zlink.samples.bingo.server.play.infrastructure.zlink.spots.bingoroomspot.handlers;

import systems.zlink.framework.spots.ZLinkSpotSubscriptionHandler;
import systems.zlink.samples.bingo.server.play.infrastructure.zlink.spots.bingoroomspot.BingoRoomSpot;
import systems.zlink.samples.bingo.shared.contracts.Messages;

public final class BingoWinnerEventHandler
    implements ZLinkSpotSubscriptionHandler<BingoRoomSpot, Messages.BingoWinnerEvent> {
    @Override
    public void handle(BingoRoomSpot spot, Messages.BingoWinnerEvent message) {
        spot.announceWinner(message);
    }
}

File: framework/languages/java/samples/kotlin/Bingo/Server/Play/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/play/infrastructure/zlink/spots/bingoroomspot/handlers/BingoWinnerEventHandler.kt
package systems.zlink.samples.kotlin.bingo.server.play.infrastructure.zlink.spots.bingoroomspot.handlers

import systems.zlink.framework.kotlin.ZLinkSuspendingSpotSubscriptionHandler
import systems.zlink.samples.kotlin.bingo.server.play.infrastructure.zlink.spots.bingoroomspot.BingoRoomSpot
import systems.zlink.samples.kotlin.bingo.shared.contracts.BingoWinnerEvent

class BingoWinnerEventHandler :
    ZLinkSuspendingSpotSubscriptionHandler<BingoRoomSpot, BingoWinnerEvent> {
    override suspend fun handle(spot: BingoRoomSpot, event: BingoWinnerEvent) {
        spot.announceWinner(event)
    }
}
